### DISCUSSION TITLE
"Unhandled Error: TypeError: Attempting to Define Property on Object That is Not Extensible in Vueform with Schema"

### DISCUSSION THREAD

**Discussion by MBucklain** [vueform/vueform#373] [Questions]
### Environment

---

#### **Frameworks and Build Tools**:
- **Frontend Framework**: Vue 3.5.13
- **Backend Framework**: Laravel 10.x (or your specific Laravel version)
- **Build Tool**: Laravel Mix 6.x (Webpack)  
  _(or Vite if you've migrated to it in Laravel 9/10)_

#### **Vueform Integration**:
- **Vueform Theme**: Tailwind CSS (with custom styles and Vueform CSS)  
  _(Vueform styles imported from `@vueform/vueform/dist/vueform.css` alongside Tailwind)_  

#### **CSS Framework**:
- **CSS Preprocessor**: PostCSS (via Laravel Mix)
- **CSS Framework**: Tailwind CSS 3.x

#### **Package Manager**:
- **npm**: 8.x.x
- **Alternative PMs**: Tested with Yarn (v1.22.x) in some cases.

#### **Laravel-Specific Environment**:
- **Laravel Version**: Laravel 10.x (or specify your version)
- **PHP Version**: 8.1.x or higher
- **Database**: MySQL 8.x or PostgreSQL (depending on your setup)
- **Build Configuration**:
  - Laravel Mix Webpack (`webpack.mix.js`) example:
    ```javascript
    const mix = require('laravel-mix');

    mix.js('resources/js/app.js', 'public/js')
       .postCss('resources/css/app.css', 'public/css', [
         require('tailwindcss'),
       ])
       .vue();
    ```

---

### **Development Environment**
- **OS**: Ubuntu 22.04 (Local Development)
- **Browser**: Chrome 116.x (latest), Firefox (latest)
- **Node.js**: 16.x.x
- **Database**: MySQL 8.x

---

If additional Laravel-specific configuration details are needed, such as `webpack.mix.js`, `vite.config.js`, or Blade templates for the frontend integration, I can provide them as well!

### Reproduction

Reproduction Repo: [vueform-repro](https://github.com/MBucklain/vueform-repro)

### **Bug Description**

When using Vueform in a Vue 3 application, the `schema` object passed to the `vue-form` component causes an error due to conflicts with Vue's reactivity system. Even after ensuring that the `schema` is a plain JavaScript object (using `markRaw` or `Object.freeze`), Vueform fails with the following error:

```
Unhandled Promise Rejection: TypeError: Attempting to define property on object that is not extensible.
```

This issue occurs even when passing a minimal `schema` directly to the `vue-form` component or using a dynamically initialized schema marked as raw.

---

### **Expected Behavior**
The Vueform component should accept a plain JavaScript `schema` object and render the form fields without errors. Vueform should not attempt to modify non-extensible or raw objects.

---

### **Observed Behavior**
1. The `schema` object is wrapped as a `Proxy` by Vue's reactivity system (`__v_skip: true`), even when marked as raw.
2. Vueform fails with the error: **"Attempting to define property on object that is not extensible."**
3. The issue persists regardless of whether the schema is:
   - Passed inline in the template.
   - Marked as raw using `markRaw`.
   - Cloned to ensure non-reactivity.

---

### **Error Logs**
1. **Initialization Logs**:
   ```plaintext
   [Log] Schema initialized: – Proxy {
       name: { label: "Name", type: "text", rules: ["required"] },
       email: { label: "Email", type: "email", rules: ["required", "email"] },
       submit: { type: "button", buttonLabel: "Submit", submits: true }
   }
   ```

2. **Console Errors**:
   ```plaintext
   [Vue warn]: Unhandled error during execution of setup function
   at <Vueform schema= {name: {label: "Name", type: "text", rules: ["required"]}, ...}>
   at <DoctorRegistrationForm>
   at <App>
   Unhandled Promise Rejection: TypeError: Attempting to define property on object that is not extensible.
   ```

---

Let me know if this description is clear or if additional details are needed!

### **Additional Context**

The issue seems to arise from Vue's reactivity system wrapping the `schema` object as a `Proxy`, causing Vueform to fail when attempting to modify it. This persists even when the schema is marked as raw or passed as a plain JavaScript object.

### Logs

```shell-script
### **Logs**


[Log] Schema initialized: – Proxy {name: {label: "Name", type: "text", rules: ["required"]}, email: {label: "Email", type: "email", rules: ["required", "email"]}, submit: {type: "button", buttonLabel: "Submit", submits: true}}

[Warning] [Vue warn]: Unhandled error during execution of setup function
  at <Vueform schema= {name: {label: "Name", type: "text", rules: ["required"]}, ...}>
  at <DoctorRegistrationForm>
  at <App>

[Error] Unhandled Promise Rejection: TypeError: Attempting to define property on object that is not extensible.
  logError (app.js:10112)
  handleError (app.js:10099)
  callWithErrorHandling (app.js:10045)
  flushJobs (app.js:10248)
```

**Comment by adamberecz** [comment]
Thank you for the detailed issue.

It seems like that Vueform plugin is [not installed](https://github.com/MBucklain/vueform-repro/blob/main/resources/js/app.js).

Please follow the [installation steps](https://vueform.com/docs/installation#manual-installation) or use the [installation wizard](https://vueform.com/docs/installation#installation) to get a basic setup. After that I'd recommend checking out how to [render forms](https://vueform.com/docs/rendering-forms) properly.
